### PR TITLE
Use Liquibase for database migrations

### PR DESCRIPTION
**Pull request by VithikaS**
### Description

This PR introduces liquibase for database migrations. Base line for schema is v5.2.0. All subsequent ddl changes to database will be applied using liquibase changesets.

### Addressed Issue

Closes https://github.com/DependencyTrack/hyades/issues/774

### Checklist

- [x] I have read and understand the [contributing guidelines](../CONTRIBUTING.md#pull-requests)
- [ ] This PR fixes a defect, and I have provided tests to verify that the fix is effective
- [x] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- [ ] This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly

### CODE REVIEW

File: src/main/resources/migration/changelog-v5.3.0.xml [hunk 1]
```diff
@@ -0,0 +1,2298 @@
...
+            </column>
+            <column name="PROPERTYTYPE" type="VARCHAR(255)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="PROPERTYVALUE" type="VARCHAR(1024)"/>
+        </createTable>
+
+        <createIndex indexName="WORKFLOW_STATE_N49" tableName="WORKFLOW_STATE">
+            <column name="PARENT_STEP_ID"/>
+        </createIndex>
```
Comment (nscuro) [inline src/main/resources/migration/changelog-v5.3.0.xml:1246]: There are still some indexes here with `_N[0-9]+` notation. We should fix those names as well.

File: src/main/resources/migration/changelog-v5.3.0.xml [hunk 2]
```diff
@@ -0,0 +1,2298 @@
...
+            <column name="locked_at" type="TIMESTAMP WITHOUT TIME ZONE">
+                <constraints nullable="false"/>
+            </column>
+            <column name="locked_by" type="VARCHAR(255)">
+                <constraints nullable="false"/>
+            </column>
+        </createTable>
+
+        <addForeignKeyConstraint baseColumnNames="VULNERABILITY" baseTableName="AFFECTEDVERSIONATTRIBUTION"
+                                 constraintName="AFFECTEDVERSIONATTRIBUTION_FK1" deferrable="true"
```
Comment (nscuro) [inline src/main/resources/migration/changelog-v5.3.0.xml:1942]: Only looking at this just now. Should we also make the foreign key names more predictable? I am not sure how consistent DataNucleus' naming is / was.

**Review by nscuro — COMMENTED** [review]
Can't wait for this to get merged, this will be huge for all development going forward :)

File: src/main/resources/migration/changelog-v5.3.0.xml (hunk 1 above)
Comment (VithikaS) [reply src/main/resources/migration/changelog-v5.3.0.xml:1246]: hawk eye :D  .. and i couldn't spot it even after creating a spreadsheet for it. Fixing now

File: src/main/resources/migration/changelog-v5.3.0.xml (hunk 2 above)
Comment (VithikaS) [reply src/main/resources/migration/changelog-v5.3.0.xml:1942]: We can rename it to CurrentTable_ReferencedTable_FK. 
Found this query somewhere to give us all constraint information we need 

`SELECT
    tc.table_schema, 
    tc.constraint_name, 
    tc.table_name, 
    kcu.column_name, 
    ccu.table_schema AS foreign_table_schema,
    ccu.table_name AS foreign_table_name,
    ccu.column_name AS foreign_column_name 
FROM information_schema.table_constraints AS tc 
JOIN information_schema.key_column_usage AS kcu
    ON tc.constraint_name = kcu.constraint_name
    AND tc.table_schema = kcu.table_schema
JOIN information_schema.constraint_column_usage AS ccu
    ON ccu.constraint_name = tc.constraint_name
WHERE tc.constraint_type = 'FOREIGN KEY';`

**Review by nscuro — APPROVED** [review]
Yay!

**Pull request merged**